### PR TITLE
Android back button functionality and some style changes

### PR DESCRIPTION
**Pull request by adrianczuczka**
Big changes: Mostly to the clickable list of tours in the header. First, restructured the code so that the OK button in the header actually changes the tour, rather than just pressing a tour in the list. This allows the user to press the back button (on Android) if they choose a tour in the list, and then decide against actually changing it, and gives a use to the OK button, which was previously only there show/hide the tour list interface. Also made the entire tour list interface smaller, replacing unused whitespace with an opaque background. Clicking this opaque part is the same as the back button. Shown below is a comparison of the changes.
screenshot_2018-05-09-11-40-56-827_uk ac edina curiousedinburghscreenshot_2018-05-09-10-34-09-952_uk ac edina curiousedinburgh

Next change is in the list of tour places. Before, holding your finger on one of the places would make it go black: a feature of TouchableHighlight. Changed this to have no effect when holding your finger on a tour place. Also moved the places' respective numbers to the title header to use space better. When clicking on a tour place, there is now a slide up/down animation, and the X button is replaced with a down arrow. This is a comparison of the old tour list versus the new one:
screenshot_2018-05-09-11-51-14-586_uk ac edina curiousedinburghscreenshot_2018-05-09-12-08-30-430_uk ac edina curiousedinburgh
And the old tour list place versus the new one:

screenshot_2018-05-09-11-51-22-878_uk ac edina curiousedinburghscreenshot_2018-05-09-12-18-55-677_uk ac edina curiousedinburgh

### CODE REVIEW

**Comment by jollopre** [comment]
@adrianczuczka please solve the conflicting files first and after I'll quickly look at the PR, ok?

**Comment by adrianczuczka** [comment]
Sorry, I forgot to mention that there will be conflicts if one of the pull requests is accepted. I'm not at work right now but I'll solve the conflicts first thing tomorrow morning, is that ok?

**Comment by adrianczuczka** [comment]
I've had major problems running the app today, so I wasn't able to fix the conflicts yet. The problem is that apparently the native code and JavaScript code is out of sync, so I'm trying to update react-native right now.

**Comment by adrianczuczka** [comment]
32665295_10211121495302472_1528173089316143104_n
This is what the error looks like, and it seems quite hard to fix. This happened before I changed, so the exact same code worked a week ago.

**Comment by adrianczuczka** [comment]
Update: the app now works again after updating react-native to the latest version, even though I did exactly that 5 minutes ago already and nothing changed. I feel like I'm going crazy.

**Comment by adrianczuczka** [comment]
@jollopre the branch should finally be ready to look at. The amount of commits is due to some duplicates and rebasing, so it looks like it's more than it actually is. Unfortunately, the already applied style changes seem to still be reflected in the changed files, but the main changes are still there to see.

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "html-entities": "1.2.1",
     "prop-types": "15.5.10",
     "react": "16.0.0-alpha.12",
-    "react-native": "0.47.2",
+    "react-native": "0.55.4",
```
Comment (jollopre) [inline package.json]: that's quite a big change. Did you check if everything works well afterwards?

**Comment by jollopre** [comment]
@adrianczuczka I think is best if we sit together to review this PR because I find it hard to distinguish what this PR is fixing, ok?

File: package.json (hunk 1 above)
Comment (adrianczuczka) [reply package.json]: I realize it is quite a big change, I wouldn't have done it if I didn't get that error I was talking about before. I did try it on my phone, and it worked as normal.

**Comment by adrianczuczka** [comment]
No problem, I'm happy to do that.

**Comment by jollopre** [comment]
I'm going to close this issue since we already merged a cleaner history

**Pull request closed without merging**